### PR TITLE
Add Databricks config to Spark config

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
This config is used by the dbx plugin in propeller to submit the dbx job request.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

File: protos/flyteidl/plugins/spark.proto [hunk 1]
```diff
@@ -21,4 +21,8 @@ message SparkJob {
     map<string, string> sparkConf = 4;
     map<string, string> hadoopConf = 5;
     string executorPath = 6; // Executor path for Python jobs.
+    // databricksConf is base64 encoded string which stores databricks job configuration.
```
Comment (wild-endeavor) [inline protos/flyteidl/plugins/spark.proto:24]: why is the databricks one a b64 encoded string while the spark and hadoop conf's above just a map of str, str?  Is the databricks one a binary object that needs to be encoded?

**Review by wild-endeavor — APPROVED** [review]
approve but maybe address question in a comment?

File: protos/flyteidl/plugins/spark.proto (hunk 1 above)
Comment (pingsutw) [reply protos/flyteidl/plugins/spark.proto:24]: Because it's nested dict. wait, I think I can use `Struct` here. Let me update it

Comment (tmathur-hbo) [reply protos/flyteidl/plugins/spark.proto:24]: +1

**Review by wild-endeavor — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]

**Pull request merged**